src/module_4/fit.py

```diff
@@ -0,0 +1,77 @@
+import logging
+import pickle
+import json
+from datetime import datetime
+from pathlib import Path
+from sklearn.base import BaseEstimator
+from push_model import Push_Model
+from utils import load_filtered_data
+
+logging.basicConfig(
+    level=logging.INFO, format="%(asctime)s - %(levelname)s - %(message)s"
+)
+
+
+def save_model(model: BaseEstimator, model_name: str) -> str:
+    """
+    Save the model to the models folder.
+
+    Args:
+        model (BaseEstimator): The trained model.
+        model_name (str): The name of the model.
+
+    Returns:
+        str: The path to the saved model.
+    """
+    model_path = Path(Path(__file__).parent, "models", f"{model_name}.pkl")
```

Comment:
Allow model_path as parameter for more flexibility and set this as default. IN additon if you prefer you can join paths or paths with strings with "/"